### ISSUE TITLE
Using FDS in ML project

### ISSUE THREAD

**Issue by thebluewind** [YyzHarry/imbalanced-regression#20]
Hello, in your paper on the problem of deep imbalance regression, I have the privilege of learning about the smoothing methods of LDS and FDS. In one of my machine learning projects predicting convective cloud precipitation, I wanted to use FDS to play a role in it because of the imbalance between non-precipitation samples and precipitation samples. I wonder how to smooth the feature statistic without knowing its label in the test set, in my data, my data is very unbalanced (70% of the data without precipitation), which results in the characteristic statistics of each label interval are particularly similar (about 98%), so that the smoothing effect is still not significant, and if there are some important points to pay attention to if using FDS in machine learning?

**Comment by YyzHarry** [comment]
Thanks for your interest.

> characteristic statistics of each label interval are particularly similar

Seems to me the network does not learn discriminative features in the first place. Maybe you can try to see whether the current model overfits the training data; if not, try to use deeper / wider model, adjust your learning parameters.

> how to smooth the feature statistic without knowing its label in the test set

FDS is not activated during inference; see #13

**Comment by w17716383006** [comment]
你好，在ML中使用标签平滑是修改模型的损失函数吗？（比如将损失函数定义为loss = weighted_mse_loss(preds, labels, weights=weights)）还是直接修改标签？如果直接修改标签的话有没有代码示例？

**Comment by YyzHarry** [comment]
See [here](https://github.com/YyzHarry/imbalanced-regression#apply-lds-and-fds-on-other-datasets--models) for examples on how to apply LDS and FDS on custom datasets and models.
